admin_manual/configuration/occ_command.rst

```diff
@@ -0,0 +1,252 @@
+administrators, and displays a status screen warning that the server is in 
+maintenance mode. Users who are not already logged in cannot log in until 
+maintenance mode is turned off. When you take the server out of maintenance mode 
+logged-in users must refresh their Web browsers to continue working::
+
+ $ sudo -u www-data php occ maintenance:mode --on
+ $ sudo -u www-data php occ maintenance:mode --off
+ 
+Putting your ownCloud server into single-user mode allows admins to log in and 
+work, but not ordinary users. This is useful for performing maintenance and 
+troubleshooting on a running server::
+
+ $ sudo -u www-data php occ maintenance:singleuser --on
+   Single user mode enabled
+   
+And turn it off when you're finished::
+
+ $ sudo -u www-data php occ maintenance:singleuser --off
+   Single user mode disabled
+
+TODO: What does  maintenance:repair do? Needs details::
+  
+  $ sudo -u www-data php occ maintenance:repair
+    - Repair mime types  
+    - Repair config
+ 
+User Commands
+-------------
+
+The ``user`` commands reset passwords, display a simple report showing how 
+many users you have, and when a user was last logged in.
+
+You can reset any user's password, including administrators (see 
+:doc:`reset_admin_password`). In this example the username is layla::
+
+ $ sudo -u www-data php occ user:resetpassword layla
+   Enter a new password: 
+   Confirm the new password: 
+   Successfully reset password for layla
+   
+View a user's most recent login::   
+   
+ $ sudo -u www-data php occ user:lastseen layla 
+   layla`s last login: 09.01.2015 18:46
+   
+TODO: does this count LDAP and other external users, or local only?::
+
+ $ sudo -u www-data php occ user:report
+   +------------------+---+
+   | User Report      |   |
+   +------------------+---+
+   | OC_User_Database | 2 |
+   |                  |   |
+   | total users      | 2 |
+   |                  |   |
+   | user directories | 3 |
+   +------------------+---+
+   
+Apps Commands
+-------------
+
+The ``app`` commands list, enable, and disable apps. This lists all of your 
+installed apps, and shows whether they are enabled or disabled::
+
+ $ sudo -u www-data php occ app:list
+ 
+Enable an app::
+
+ $ sudo -u www-data php occ app:enable external
+   external enabled
+   
+Disable an app::
+
+ $ sudo -u www-data php occ app:disable external
+   external disabled
+   
+Upgrade Command
+---------------
+
+When you are performing an update or upgrade on your ownCloud server, it is 
+better to use ``occ`` to perform the database upgrade step, rather than the Web 
+GUI,  in order to avoid timeouts.
+
+TODO: what timeouts? What causes them?
+
+You can perform a dry-run first to see what will happen, without changing 
+anything::
+
+ $ sudo -u www-data php occ upgrade --dry-run
+
+When this looks satisfactory, you can go ahead and perform the upgrade::
+
+ $ sudo -u www-data php occ upgrade
+ 
+TODO why would you want to use --skip-migration-test? ::
```

Comment:
ownCloud updates first try to upgrade with a simulation, that means that all whole database tables are copied to a new place and then the DB update is simulated on that. That ensures that the oC DB won't do faulty updates. (that's also what the dry-run does)

However, this means that the update will obviously takes twice as much time thus you can skip it in case of large installations where it might take ages otherwise. (ages as in "potentially days")